### PR TITLE
Next.js App Router Example

### PR DESCRIPTION
**Pull request by feedthejim**

### CODE REVIEW

File: examples/with-next/next.config.mjs [hunk 1]
```diff
@@ -1,30 +1,4 @@
 /** @type {import('next').NextConfig} */
```
Comment (kettanaito) [inline examples/with-next/next.config.mjs:1]: I suppose the config file can be deleted.

File: examples/with-next/tsconfig.json [hunk 2]
```diff
@@ -19,7 +19,8 @@
     ],
     "paths": {
       "@/*": ["./*"]
-    }
+    },
+    "target": "ES2017"
```
Comment (kettanaito) [inline examples/with-next/tsconfig.json:23]: Does this enable any particular feature?

File: examples/with-next/app/layout.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import type { Metadata } from 'next'
 import { Inter } from 'next/font/google'
-import { MockProvider } from './mockProvider'
+import { MSWProvider } from './msw-provider'
```
Comment (kettanaito) [inline examples/with-next/app/layout.tsx:3]: While I think it a noble idea to collocate the setup within the same component, it yields more files to create for the end user. 

Your initial setup was better, in my opinion.

- server setup inline in `layout.tsx`;
- `MockProvider` as a single copy-pastable file you import in `layout.tsx` and wrap your children in it.

File: examples/with-next/tsconfig.json (hunk 2 above)
Comment (feedthejim) [reply examples/with-next/tsconfig.json:23]: I need to remove all the noise, I just added it by reflex

File: examples/with-next/app/layout.tsx (hunk 3 above)
Comment (kettanaito) [reply examples/with-next/app/layout.tsx:3]: I really like how you utilize `layout.tsx` as both client-side and server-side entrypoints!

**Pull request merged**